#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_fix_emacs-command-line.dpatch by Youhei SASAKI <uwabami@gfd-dennou.org>
##
## DP: fix handling commandline arguments, see elscreen-ml
## DP: http://www.heimat.gr.jp/pipermail/elips/2010/001976.html

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' elscreen-1.4.6~/elscreen.el elscreen-1.4.6/elscreen.el
--- elscreen-1.4.6~/elscreen.el	2010-06-04 01:07:56.000000000 +0900
+++ elscreen-1.4.6/elscreen.el	2010-06-04 01:08:57.000000000 +0900
@@ -1691,6 +1691,8 @@
                   ("-e"                . elscreen-command-line-funcall))))
 
   (static-when elscreen-on-emacs
+    (if (string-match "^\\(19\\|2[0-2]\\)" emacs-version) ; emacs22 or prior-to
+	(progn
     (defun elscreen-e21-command-line ()
       (when (string-match "\\`-" argi)
         (error "Unknown option `%s'" argi))
@@ -1708,6 +1710,27 @@
     (add-hook 'after-init-hook (lambda ()
                                  (add-to-list 'command-line-functions
                                               'elscreen-e21-command-line t))))
+      ; else
+      (progn
+    (defun elscreen-e23-command-line ()
+      (when (string-match "\\`-" argi)
+        (error "Unknown option `%s'" argi))
+      (setq file-count (1+ file-count))
+      (setq inhibit-startup-buffer-menu t)
+      (let* ((file
+              (expand-file-name
+               (command-line-normalize-file-name orig-argi)
+               cl1-dir)))
+        (elscreen-command-line-find-file file file-count cl1-line cl1-column))
+      (setq cl1-line 0)
+      (setq cl1-column 0)
+      t) ; defun
+
+    (add-hook 'after-init-hook (lambda ()
+                                 (add-to-list 'command-line-functions
+                                              'elscreen-e23-command-line t))))
+      ); endif
+    )
 
   (static-when elscreen-on-xemacs
     (defadvice command-line-1 (around elscreen-xmas-command-line-1 activate)
